spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkAppenderFactory.java

```diff
@@ -161,29 +164,37 @@ private StructType lazyPosDeleteSparkType() {
   }
 
   @Override
-  public FileAppender<InternalRow> newAppender(OutputFile file, FileFormat fileFormat) {
+  public FileAppender<InternalRow> newAppender(OutputFile outputFile, FileFormat format) {
+    return newAppender(
+        EncryptedFiles.encryptedOutput(outputFile, (EncryptionKeyMetadata) null), format);
```

Comment:
Can you add a pass-through factory method to `EncryptedFiles` rather than constructing one here? I think it would be better to call `EncryptedFiles.plainAsEncryptedOutput` rather than passing null key metadata.